daemon/api_model.go

```diff
@@ -69,3 +89,96 @@ func postModel(c *Command, r *http.Request, _ *auth.UserState) Response {
 	return AsyncResponse(nil, &Meta{Change: chg.ID()})
 
 }
+
+// helper for parsing url options into formatting option vars
+func parseHeadersFormatOptionsFromURL(url *url.URL) (
+	jsonResult bool,
+	headersOnly bool,
+	headers map[string]string,
+	err error,
+) {
+	headers = map[string]string{}
+	q := url.Query()
+	for k := range q {
+		if k == "json" {
+			switch q.Get(k) {
+			case "false":
+				jsonResult = false
+			case "headers":
+				headersOnly = true
+				fallthrough
+			case "true":
+				jsonResult = true
+			default:
+				err = errors.New(`"json" query parameter when used must be set to "true" or "headers"`)
+				return
+			}
+			continue
+		}
+		headers[k] = q.Get(k)
+	}
+
+	return
+}
+
+// getModel gets the current model assertion using the DeviceManager
+func getModel(c *Command, r *http.Request, _ *auth.UserState) Response {
+	jsonResult, headersOnly, _, err := parseHeadersFormatOptionsFromURL(r.URL)
+	if err != nil {
+		return BadRequest(err.Error())
+	}
+
+	devmgr := c.d.overlord.DeviceManager()
+	if devmgr == nil {
+		return InternalError("accessing device manager failed")
+	}
+
+	model, err := devmgr.Model()
+	if err != nil {
+		return InternalError("accessing model failed: %v", err)
+	}
+
+	if jsonResult {
+		modelJSON := modelAssertJSONResponse{}
+
+		modelJSON.Headers = model.Headers()
+		if !headersOnly {
+			modelJSON.Body = string(model.Body())
+		}
+
+		return SyncResponse(modelJSON, nil)
+	}
+
+	return AssertResponse([]asserts.Assertion{model}, true)
+}
+
+// getSerial gets the current serial assertion using the DeviceManager
+func getSerial(c *Command, r *http.Request, _ *auth.UserState) Response {
+	jsonResult, headersOnly, _, err := parseHeadersFormatOptionsFromURL(r.URL)
+	if err != nil {
+		return BadRequest(err.Error())
+	}
+
+	devmgr := c.d.overlord.DeviceManager()
+	if devmgr == nil {
```

Comment:
this should happen at all at this point (not sure we do this kind of checks in other places)

Reply:
Removed the check here in that case